### PR TITLE
CFD detailed modal, `Toggle`, `TradingInput` and `TradeTab`

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP
- [x]  051-PositionDetailsModal
- [x] 107-TradingInput
- [x] 044-TradeTab
- [x] 106-Toggle
- [ ] 036-TradingHeader
### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component: 0
- new loop: 0
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)

### CODE REVIEW

**Review by Luphia — APPROVED** [review]
- no bug found

File: src/components/position_details_modal/position_details_modal.tsx [hunk 1]
```diff
@@ -64,21 +71,111 @@ const PositionDetailsModal = ({
...
+
+  const buttonClickHandler = () => {
+    // console.log('btn clicked');
+
+    let changedProperties = {};
+    let takeProfit = {};
+    let stopLoss = {};
+
+    // Detect if tpValue has changed
+    if (takeProfitToggle && takeProfitValue !== openCfdDetails.takeProfit) {
```
Comment (arealclimber) [inline src/components/position_details_modal/position_details_modal.tsx:120]: - Remove `toggle` properties, unnecessary to pass `toggle` to Metamask
- Turn off `takeProfitToggle`, pass `takeProfitAmount=0` to cancel the take-profit

File: src/components/position_details_modal/position_details_modal.tsx [hunk 2]
```diff
@@ -64,21 +71,111 @@ const PositionDetailsModal = ({
...
+    if (guaranteedChecked !== openCfdDetails.guranteedStop) {
+      stopLoss = {stopLossToggle: stopLossToggle, stopLossAmount: stopLossValue};
+      changedProperties = {
+        ...changedProperties,
+        guranteedStopChecked: guaranteedChecked,
+        stopLoss,
+      };
+    }
+
+    if (Object.keys(changedProperties).length > 0) {
```
Comment (arealclimber) [inline src/components/position_details_modal/position_details_modal.tsx:159]: - If no changed properties, restrict to submit

File: src/components/trade_tab/trade_tab.tsx [hunk 3]
```diff
@@ -79,19 +128,26 @@ const TradeTab = () => {
...
+  const isDisplayedMarginSize = marginLength > 7 ? 'text-sm' : 'text-base';
+  const isDisplayedValueSize = valueOfPositionLength > 7 ? 'text-sm' : 'text-base';
+  const isDisplayedDividerSpacing =
+    valueOfPositionLength > 10 || marginLength > 10 ? 'top-430px' : 'top-420px';
 
   // ----------margin area----------
   const displayedMarginSetting = (
     <TradingInput
+      getInputValue={getMarginInputValue}
       lowerLimit={0}
```
Comment (arealclimber) [inline src/components/trade_tab/trade_tab.tsx:147]: - upperLimit = 10^9
- #192

**Pull request merged**